File: C++/A/26A-almostprime.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n, amount = 0, pfactors[3001] = {0};
    cin >> n;
    bool b[3001] = {false};
    for (int i = 2; i <= n; ++i)
    {
        if (!b[i])
            for (int j = i * 2; j <= n; j += i)
            {
                b[j] = true;
                pfactors[j]++;
            }
        (pfactors[i] == 2) ? amount++ : 0;
    }
    cout << amount;
}

File: Python/26A-almostprime.py
n = int(input())
amount = 0
b = [False] * (n+1)
pfactors = [0] * (n+1)

for i in range(2, n+1):
    if not b[i]:
        for j in range(i * 2, n+1, i):
            b[j] = True
            pfactors[j] += 1

    if pfactors[i] == 2:
        amount += 1

print(amount)
